### ISSUE TITLE
Reactive systems example won't compile

### ISSUE THREAD

**Issue by mhaemmerle** [JuDelCo/Entitas-Cpp#2]
When trying to add the reactive system example from the readme to the project's main.cpp, Visual Studio won't compile with the following error:

> Error C2280 'EntitasPP::IReactiveSystem::IReactiveSystem(void)': attempting to reference a deleted function

When omitting the constructor, CreateSystem<FooSystem>() will fail because it apparently expects one.

Assigning the trigger event in the constructor also fails as the expected and returned type don't match.

Any ideas?

Here's a gist: https://gist.github.com/mhaemmerle/851273780b3af155d801302458e24b16

**Comment by JuDelCo** [comment]
Hi, unfortunately I don't have time to maintain the lib at this moment because of my Job.

I never used Visual Studio (I use GCC on Linux) so I can't (if i have time) reproduce your problem.

By the way, I want to warn you that the current state of this library is a bit buggy in terms of memory management (memory leaks!).

I hope i can retake this in the next months and fix all the bugs.

Thanks for your interest, really !

**Comment by mhaemmerle** [comment]
Thanks for the reply!

I'am currently trying it out together with Unreal Engine 4 in a toy project as the other ECS (anax, entityx, Artemis) don't have reactive systems, are slow or very verbose.

I've found a thread on StackOverflow that it could be related to a bug in Visual Studio 2015, but can't verify that. I'll investigate futher :)

Can you reply to this thread when you're working on Entitas-Cpp again?

This might interest you: http://tilemapkit.com/2015/10/entity-component-systems-compared-benchmarked-entityx-anax-artemis/

Cheers

**Comment by JuDelCo** [comment]
Yes, no problem. I will notify in here when back on track again ;)

**Comment by JuDelCo** [comment]
Hi, this was fixed in the latest commit bf5653e8bf41b06f23056876756dd7334d6d211b

```cpp
class FooSystem : public IReactiveSystem {
	public:
		FooSystem() {
			trigger = Matcher_AllOf(DemoComponent).OnEntityAdded();
		}

		void Execute(std::vector<EntityPtr> entities) {
			for (auto &e : entities) {
			}
		}
};
```
